framework/wazuh/core/utils.py

```diff
@@ -345,6 +350,24 @@ def get_nested_fields(dikt, select_field):
     return result_list
 
 
+def filter_by_resource_type(array: typing.List, resource_type: str) -> typing.List:
+    """Filter a list of dictionaries to return only the elements that match with the specified resource_type value.
+
+     Parameters
+     ----------
+     array : list
+         Array of elements. It contains all the results without any filter.
+     resource_type: str
+         The resource_type to filter by.
+
+     Returns
+     -------
+     result_list : list
+         Filtered array of dicts with only the selected (and required) fields as keys.
+     """
+    return [x for x in array if x['resource_type'] == resource_type]
+
+
```

Comment:
Was this necessary? Maybe we could use the `q` param. This feels too specific. `process_array` is used in a lot of endpoints.

Reply:
After further analysis I have removed this from utils. Instead, the filtering logic is now placed inside the `security` module, to apply it before the `process_array` call.